contracts/cosmwasm-vm/cw-cluster-connection/src/state.rs

```diff
@@ -0,0 +1,180 @@
+    }
+
+    pub fn get_relayer(&self, store: &dyn Storage) -> Result<Addr, ContractError> {
+        Ok(self.relayer.load(store)?)
+    }
+
+    pub fn store_conn_sn(&mut self, store: &mut dyn Storage, sn: u128) -> StdResult<()> {
+        self.conn_sn.save(store, &sn)?;
+        Ok(())
+    }
+
+    pub fn get_next_conn_sn(&self, store: &mut dyn Storage) -> Result<u128, ContractError> {
+        let mut connsn = self.conn_sn.load(store).unwrap_or(0);
+        connsn += 1;
+        self.conn_sn.save(store, &connsn)?;
+        Ok(connsn)
+    }
+
+    pub fn store_fee(
+        &mut self,
+        store: &mut dyn Storage,
+        to: NetId,
+        message_fee: u128,
+        response_fee: u128,
+    ) -> StdResult<()> {
+        self.message_fee.save(store, to.clone(), &message_fee)?;
+        self.response_fee.save(store, to, &response_fee)?;
+        Ok(())
+    }
+    pub fn get_message_fee(&self, store: &dyn Storage, to: NetId) -> u128 {
+        self.message_fee.load(store, to).unwrap_or(0)
+    }
+
+    pub fn get_response_fee(&self, store: &dyn Storage, to: NetId) -> u128 {
+        self.response_fee.load(store, to).unwrap_or(0)
+    }
+
+    pub fn store_receipt(
+        &mut self,
+        store: &mut dyn Storage,
+        src_network: NetId,
+        connsn: u128,
+    ) -> StdResult<()> {
+        self.receipts
+            .save(store, (src_network.to_string(), connsn), &true)?;
+        Ok(())
+    }
+
+    pub fn get_receipt(&self, store: &dyn Storage, src_network: NetId, sn: u128) -> bool {
+        self.receipts
+            .load(store, (src_network.to_string(), sn))
+            .unwrap_or(false)
+    }
+
+    pub fn store_denom(&mut self, store: &mut dyn Storage, denom: String) -> StdResult<()> {
+        self.denom.save(store, &denom)?;
+        Ok(())
+    }
+
+    pub fn get_denom(&self, store: &dyn Storage) -> String {
+        self.denom.load(store).unwrap()
+    }
+
+    pub fn store_validator(&mut self, store: &mut dyn Storage, validator: String) -> StdResult<()> {
+        self.validators.save(store, validator, &true)?;
+        Ok(())
+    }
+
+    pub fn remove_validator(
+        &mut self,
+        store: &mut dyn Storage,
+        validator: String,
+    ) -> StdResult<()> {
+        self.validators.remove(store, validator);
+        Ok(())
+    }
+
+    pub fn clear_validators(&mut self, store: &mut dyn Storage) -> StdResult<()> {
+        self.validators.clear(store);
+        Ok(())
+    }
+
+    pub fn get_validators(&self, store: &dyn Storage) -> StdResult<Vec<String>> {
+        let mut validators_list: Vec<String> = Vec::new();
+        let validators_iter =
+            self.validators
+                .range(store, None, None, cosmwasm_std::Order::Ascending);
+
+        for item in validators_iter {
+            let (validator_addr, is_active) = item?;
+            if is_active {
+                validators_list.push(validator_addr);
+            }
+        }
+
+        Ok(validators_list)
```

Comment:
converting validators map to list and using that to lookup in signature verification? cant signature verification lookit up directly in the storage? if we dont want to touch storage we can return hashmap from here as well.